types/node/events.d.ts

```diff
@@ -639,3 +639,163 @@ declare module 'node:events' {
     import events = require('events');
     export = events;
 }
+
+// NB: The Event / EventTarget / EventListener implementations below were copied
+// from lib.dom.d.ts, then edited to reflect Node's documentation at
+// https://nodejs.org/api/events.html#class-eventtarget.
+// Please read that link to understand important implementation differences.
+
+// For now, these Events are contained in a temporary module to avoid conflicts
+// with their DOM versions in projects that include both `lib.dom.d.ts` and `@types/node`
+declare module 'node:dom-events' {
```

Comment:
is this specific bit (dom-events) required for this PR? that one will expose `node:dom-events` for autoimports, failing at runtime (unknown builtin module)

Reply:
The plan was, if the module's exports live entirely in type-space, the `import` statements should never make it into the emit.  Looking back over it, that isn't actually the case -- the fictional module declares `const Event` and `const EventTarget`, but of course `import {EventTarget} from "node:dom-events"; new EventTarget();` would fail.  I'll try to come up with another approach.

Also: I just checked and `NodeEventTarget` is defined in the real `events` module... but *never referenced anywhere else* -- no other Node API uses this interface.  I guess the idea is for third party libraries to inherit from it?  But the interface in the current code is missing most of the properties and methods.  Will look at moving this into the real module.

Reply:
Update on the NodeEventTarget situation: that class appears in the docs but [is not actually available at runtime](https://github.com/nodejs/node/issues/39903), so I just commented the whole thing out.  It should be easy to add it back if the linked issue gets resolved.

Reply:
The changes I'm pushing now should address the events-module problems, I think.  I realized that 1) if the module is not intended for end-users to consume, it should be OK to make intermediate / alias types, and 2) I can use the same `typeof globalThis extends ....` construct to conditionally define an alias, then declare the global interface augmentation as `extends __Alias`.  As a result, there should be a global `Event` type *and* constructor, and both "sides" should reflect either the Node version or the DOM one (but never both).

I'm not sure this is "done", though.

First, it worked for Event and EventTarget because they are defined as interfaces on the type side with a `var Event` that defines the constructor side.  It wouldn't work the same way with class declarations, though in fairness I don't think many of the current globals are declared as `class`es.

Second, it necessarily involves creating extra modules and/or types that aren't intended for public use.  I've always been a little uncomfortable working with module definitions -- what makes it "ambient"?  When does this interface get exported even though I didn't say "export"?  How am I importing from a module, inside that same module? etc etc -- so I feel like if I just understood it a bit better, I could more effectively hide the "work in progress" types and whatnot.  (Suggestions are more than welcome!)

Reply:
IMO it works now, nothing can be imported, so it impacts only global symbols